Mathlib/Topology/Category/Profinite/Nobeling.lean

```diff
@@ -0,0 +1,1556 @@
+  simp only [πs'_apply, LocallyConstant.toFun_eq_coe, continuous_projRestricts,
+    LocallyConstant.coe_comap]
+
+theorem injective_πs' {o₁ o₂ : Ordinal} (h : o₁ ≤ o₂) : Function.Injective (πs' C h) :=
+  LocallyConstant.comap_injective _ (continuous_projRestricts _ _) (surjective_projRestricts _ _)
+
+end Maps
+
+section ProductsFactorisation
+
+namespace Products
+
+theorem lt_ord_of_lt {l m : Products I} {o : Ordinal} (h₁ : m < l)
+    (h₂ : ∀ i ∈ l.val, ord I i < o) : ∀ i ∈ m.val, ord I i < o :=
+  List.Sorted.lt_ord_of_lt (List.chain'_iff_pairwise.mp l.2) (List.chain'_iff_pairwise.mp m.2) h₁ h₂
+
+theorem eval_πs {l : Products I} {o : Ordinal} (hlt : ∀ i ∈ l.val, ord I i < o) :
+    πs C o (l.eval (π C (ord I · < o))) = l.eval C := by
+  simpa only [← LocallyConstant.coe_inj, πs_apply, continuous_projRestrict,
+    LocallyConstant.coe_comap] using evalFacProp C (ord I · < o) hlt
+
+theorem eval_πs' {l : Products I} {o₁ o₂ : Ordinal} (h : o₁ ≤ o₂)
+    (hlt : ∀ i ∈ l.val, ord I i < o₁) :
+    πs' C h (l.eval (π C (ord I · < o₁))) = l.eval (π C (ord I · < o₂)) := by
+  rw [← LocallyConstant.coe_inj, ← LocallyConstant.toFun_eq_coe]
+  simp only [πs'_apply, LocallyConstant.toFun_eq_coe, continuous_projRestricts,
+    LocallyConstant.coe_comap]
+  exact evalFacProps C (fun (i : I) ↦ ord I i < o₁) (fun (i : I) ↦ ord I i < o₂) hlt _
+
+theorem eval_πs_image {l : Products I} {o : Ordinal}
+    (hl : ∀ i ∈ l.val, ord I i < o) : eval C '' { m | m < l } =
+    (πs C o) '' (eval (π C (ord I · < o)) '' { m | m < l }) := by
+  ext f
+  simp only [Set.mem_image, Set.mem_setOf_eq, exists_exists_and_eq_and]
+  apply exists_congr; intro m
+  apply and_congr_right; intro hm
+  rw [eval_πs C (lt_ord_of_lt hm hl)]
+
+theorem eval_πs_image' {l : Products I} {o₁ o₂ : Ordinal} (h : o₁ ≤ o₂)
+    (hl : ∀ i ∈ l.val, ord I i < o₁) : eval (π C (ord I · < o₂)) '' { m | m < l } =
+    (πs' C h) '' (eval (π C (ord I · < o₁)) '' { m | m < l }) := by
+  ext f
+  simp only [Set.mem_image, Set.mem_setOf_eq, exists_exists_and_eq_and]
+  apply exists_congr; intro m
+  apply and_congr_right; intro hm
+  rw [eval_πs' C h (lt_ord_of_lt hm hl)]
+
+theorem head_lt_ord_of_isGood {l : Products I} {o : Ordinal}
+    (h : l.isGood (π C (ord I · < o))) (hn : l.val ≠ []) : ord I (l.val.head!) < o :=
+  prop_of_isGood C (ord I · < o) h l.val.head! (List.head!_mem_self hn)
+
+theorem isGood_mono {l : Products I} {o₁ o₂ : Ordinal} (h : o₁ ≤ o₂)
+    (hl : l.isGood (π C (ord I · < o₁))) : l.isGood (π C (ord I · < o₂)) := by
+  intro hl'
+  apply hl
+  rwa [eval_πs_image' C h (prop_of_isGood  C _ hl), ← eval_πs' C h (prop_of_isGood  C _ hl),
+    Submodule.apply_mem_span_image_iff_mem_span (injective_πs' C h)] at hl'
+
+end Products
+
+end ProductsFactorisation
+
+section Smaller
+
+namespace GoodProducts
+
+/-- The image of the `GoodProducts` for `π C (ord I · < o)` in `LocallyConstant C ℤ` -/
+def smaller (o : Ordinal) : Set (LocallyConstant C ℤ) :=
```

Comment:
why is this called "smaller"? You just started a new section -- give it a docstring just explaining in a few words what the point of this section is. The clue `smaller` is not enough for me.